src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -2239,7 +2238,9 @@ mongoc_uri_unescape (const char *escaped_string)
 
    ptr = escaped_string;
    end = ptr + len;
-   str = mcommon_string_new (NULL);
+
+   mcommon_string_append_t append;
+   mcommon_string_append_init (&append, mcommon_string_new_with_capacity ("", 0, len));
```

Comment:
```suggestion
   mcommon_string_append_init_with_capacity (&append, len);
```
Consider adding a `mcommon_string_append_init_with_capacity` given the frequency of this pattern (default initialization with custom capacity).

Reply:
Agree that a shortcut would be helpful. I'll think about a name.. I'd like it to be clear that a mcommon_string_t is being allocated.